pkg/api/v1/schema.go

```diff
@@ -19,346 +19,474 @@
 }
 
-type DiskSourceHost struct {
+type FloppyTarget struct {
+	// Device indicates the "logical" device name. The actual device name
+	// specified is not guaranteed to map to the device name in the guest OS. Treat
+	// it as a device ordering hint.
+	Device string `json:"dev"`
+	// ReadOnly
+	// Defaults to false
+	ReadOnly bool `json:"readonly,omitempty"`
+	// Tray indicates if the tray of the device is open or closed.
+	// Allowed values are "open" and "closed"
+	// Defaults to closed
+	// +optional
+	Tray TrayState `json:"tray,omitempty"`
+}
+
+// TrayState indicates if a tray of a cdrom or floppy is open or closed
+type TrayState string
+
+const (
+	// TrayStateOpen indicates that the tray of a cdrom or floppy is open
+	TrayStateOpen TrayState = "open"
+	// TrayStateClosed indicates that the tray of a cdrom or floppy is closed
+	TrayStateClosed TrayState = "closed"
+)
+
+type CDRomTarget struct {
+	// Device indicates the "logical" device name. The actual device name
+	// specified is not guaranteed to map to the device name in the guest OS. Treat
+	// it as a device ordering hint.
+	Device string `json:"dev"`
+	// ReadOnly
+	// Defaults to true
+	ReadOnly *bool `json:"readonly,omitempty"`
+	// Tray indicates if the tray of the device is open or closed.
+	// Allowed values are "open" and "closed"
+	// Defaults to closed
+	// +optional
+	Tray TrayState `json:"tray,omitempty"`
+}
+
+// Volume represents a named volume in a vm.
+type Volume struct {
+	// Volume's name.
+	// Must be a DNS_LABEL and unique within the vm.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
 	Name string `json:"name"`
-	Port string `json:"port,omitempty"`
-}
-
-// END Disk -----------------------------
-
-// BEGIN Serial -----------------------------
-
-type Serial struct {
-	Type   string        `json:"type"`
-	Target *SerialTarget `json:"target,omitempty"`
-}
-
-type SerialTarget struct {
-	Port *uint `json:"port,omitempty"`
-}
-
-// END Serial -----------------------------
-
-// BEGIN Console -----------------------------
-
-type Console struct {
-	Type   string         `json:"type"`
-	Target *ConsoleTarget `json:"target,omitempty"`
-}
-
-type ConsoleTarget struct {
-	Type *string `json:"type,omitempty"`
-	Port *uint   `json:"port,omitempty"`
-}
-
-// END Serial -----------------------------
-
-// BEGIN Inteface -----------------------------
-
+	// VolumeSource represents the location and type of the mounted volume.
+	// Defaults to Disk, if no type is specified
+	VolumeSource `json:",inline"`
+}
+
+// Represents the source of a volume to mount.
+// Only one of its members may be specified.
+type VolumeSource struct {
+	// ISCSI represents an ISCSI Disk resource that is attached to a
+	// kubelet's host machine and then exposed to the pod.
+	// More info: https://releases.k8s.io/HEAD/examples/volumes/iscsi/README.md
+	// +optional
+	ISCSI *v1.ISCSIVolumeSource `json:"iscsi,omitempty"`
```

Comment:
This ISCSI logic is tricky right now. Your comment reflects our desired use of ISCSI, but that isn't the reality right now.

The k8s ISCSIVolumeSource object contains fields we can't support right now using libvirt.  I don't have a problem with that necessarily, but we need to track what's happening here somewhere.

Reply:
Good catch. I will fix the comment for now. Once we can consume block storage, we can adjust the documentation.

Reply:
I actually wonder - what about dropping the plain iscsi support in favor of PVCs?

After all it looks like we can achieve the same level of functionality by using PVCs, or is there some gap?

Reply:
One would have to write three manifests instead of one (PV, PVC and VM), in case the storage is not provided by a storage controller. Further  that PV is not namespaces, thus a user would need access to PVs, which you can't give him. So dropping is not really an option in my eyes.

Reply:
The creation of the PV is actually a good argument, I'd expect that a user usually does not have the permissions to do this.

Reply:
Fixed the comments